compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -1052,14 +1060,14 @@ static LogicalResult generateUnifiedObject(
     llvm::outs() << "\n";
   }
 
-  ModuleOp copy = moduleOp.clone();
-  if (failed(pm.run(copy)))
-    return moduleOp.emitOpError("Failed to lower to LLVM");
+  // AIE::DeviceOp copy = deviceOp.clone();
+  if (failed(pm.run(moduleWithOneDevice)))
+    return deviceOp.emitOpError("Failed to lower to LLVM");
 
   llvm::LLVMContext llvmContext;
-  auto llvmModule = translateModuleToLLVMIR(copy, llvmContext);
+  auto llvmModule = translateModuleToLLVMIR(moduleWithOneDevice, llvmContext);
   if (!llvmModule)
-    return moduleOp.emitOpError("Failed to translate module to LLVMIR");
+    return deviceOp.emitOpError("Failed to translate module to LLVMIR");
```

Comment:
style guide nit (and I know it was here before): conditionals that don't fit on one line get braces (other @jtuyls will hunt you down)